compositor_render/src/transformations/web_renderer/chromium_sender_thread.rs

```diff
@@ -128,34 +127,46 @@ impl ChromiumSenderThread {
     fn ensure_shared_memory(
         &self,
         state: &mut ThreadState,
-        node_id: NodeId,
         resolutions: Vec<Option<Resolution>>,
     ) -> Result<(), ChromiumSenderThreadError> {
-        state.shared_memory.entry(node_id).or_default();
-
-        let frame = state.browser.main_frame()?;
-        let shared_memory = state.shared_memory.get_mut(&node_id).unwrap();
-        for (source_idx, resolution) in resolutions.into_iter().enumerate() {
-            let size = match resolution {
+        fn size_from_resolution(resolution: Option<Resolution>) -> usize {
+            match resolution {
                 Some(res) => 4 * res.width * res.height,
                 None => 1,
-            };
+            }
+        }
 
-            match shared_memory.get_mut(source_idx) {
-                Some(shmem) => {
-                    shmem.ensure_size(size, &frame)?;
-                }
-                None => {
-                    shared_memory.push(SharedMemory::new(
-                        &state.shared_memory_root_path,
-                        &node_id,
-                        source_idx,
-                        size,
-                    )?);
-                }
+        let frame = state.browser.main_frame()?;
+        let shared_memory = &mut state.shared_memory;
+
+        // Ensure size for already existing shared memory
+        for (resolution, shmem) in resolutions.iter().zip(shared_memory.iter_mut()) {
+            let size = size_from_resolution(*resolution);
+            if shmem.len() != size {
+                // TODO: This should be synchronised
+                let mut process_message = cef::ProcessMessage::new(UNEMBED_SOURCE_FRAMES_MESSAGE);
+                process_message.write_string(0, shmem.to_path_string());
+                frame.send_process_message(cef::ProcessId::Renderer, process_message)?;
+                // -----
+
+                shmem.resize(size)?;
```

Comment:
what if you only resize if shm is too small? Would this avoid potential crashes even without synchronization?

Reply:
I think it would work